Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift

```diff
@@ -25,6 +25,19 @@ public class StarknetProvider: StarknetProviderProtocol {
         self.init(starknetChainId: starknetChainId, url: url)
     }
 
+    public init(starknetChainId: StarknetChainId, url: URL, urlSession: URLSession) {
+        self.starknetChainId = starknetChainId
+        self.url = url
+        self.networkProvider = HttpNetworkProvider(session: urlSession)
+    }
+
+    public convenience init?(starknetChainId: StarknetChainId, url: String, urlSession: URLSession) {
```

Comment:
Is this really needed? Provides convenience to consumers but then they will need to an optional provider anyway.

Reply:
It's should it be there, at least for consistency